pycodestyle.py

```diff
@@ -2411,6 +2388,38 @@ def _parse_multi_options(option, split_token=','):
     return []
 
 
+def _ensure_valid_arguments(options):
+    _parse_list_multi_options(options)
+
+    options_dict = options.__dict__
+    default_types = (("verbose", int),
+                     ("quiet", int),
+                     ("repeat", bool),
+                     ("exclude", (list, tuple)),
+                     ("filename", (list, tuple)),
+                     ("select", (list, tuple)),
+                     ("ignore", (list, tuple)),
+                     ("max_line_length", int),
+                     ("format", str),
+                     ("testsuite", str),
+                     ("doctest", bool),
+                     ("reporter", object),
+                     ("config", bool),
+                     ("show_source", bool),
+                     ("show_pep8", bool),
+                     ("statistics", bool),
+                     ("count", bool),
+                     ("hang_closing", bool),
+                     ("diff", bool),
+                     ("benchmark", bool),)
+    for option, default_type in default_types:
+        if not isinstance(options_dict[option], default_type):
+            raise ValueError("The option `%s=%s` must be of the type `%s` "
```

Comment:
This is mildly better but I really don't know that we should be doing strict type-checking here.

Reply:
Then it was doing what I expected. I do not like the ideia of type checking while running on production environment to not eat CPU cycles unnecessarily.

The assert statements are meant to the development environment, and for the ones this this directly by its API, as `python-language-server`